src/main/fc/init.c

```diff
@@ -1007,28 +1009,26 @@ void init(void)
 
     setArmingDisabled(ARMING_DISABLED_BOOT_GRACE_TIME);
 
-    // On F4/F7 allocate SPI DMA streams before motor timers
-#if defined(STM32F4) || defined(STM32F7)
-#ifdef USE_SPI
+// allocate SPI DMA streams before motor timers
+#if defined(USE_SPI) && defined(USE_SPI_EARLY_INIT)
     // Attempt to enable DMA on all SPI busses
     spiInitBusDMA();
 #endif
-#endif
 
 #ifdef USE_MOTOR
     motorPostInit();
     motorEnable();
 #endif
 
-    // On H7/G4 allocate SPI DMA streams after motor timers as SPI DMA allocate will always be possible
-#if defined(STM32H7) || defined(STM32G4) || defined(AT32F435)
-#ifdef USE_SPI
+// allocate SPI DMA streams after motor timers as SPI DMA allocate will always be possible
+#if defined(USE_SPI) && defined(USE_SPI_LATE_INIT) && !defined(USE_SPI_EARLY_INIT)
```

Comment:
This is nice, but are we sure about the names `USE_SPI_LATE_INIT` / `USE_SPI_EARLY_INIT`?
Maybe something like `USE_SPI_DMA_ENABLE_EARLY` / `USE_SPI_DMA_ENABLE_LATE` would be better?

Reply:
Done :)